lib/interactive-vm.nix

```
@@ -20,7 +20,7 @@ let
   disks = lib.attrValues cfg_.disko.devices.disk;
   rootDisk = {
     name = "root";
-    file = ''"$tmp"/${(builtins.head disks).name}.qcow2'';
+    file = ''"$tmp/${(builtins.head disks).name}.qcow2"'';
```

Maybe `lib.escapeShellArg` could work better here?

---

I also agree. Also we should keep shell escaping very close to the command so we don't accidentally double quote.

---

Just for my understanding, do you mean in general or only in these concatenated instances?

It seems like it would just be the better choice in general, though it's a bit verbose unfortunately.

---

```suggestion
    file = ''"$tmp"/${lib.escapeShellArg ((builtins.head disks).name)}.qcow2'';
```

This is how I would do it, maybe move the `(builtins.head disk).name` into the `let` above as well if it seems a bit dense

---

Okay then I think I understood correctly. I updated the changes, do they look better now?